pkg/ibmvpc/sg_analysis.go

```diff
@@ -317,6 +317,19 @@ func AnalyzeSGRules(rules []*SGRule, isIngress bool) *ConnectivityResult {
 	return res
 }
 
+func mapAndAnalyzeSGRules(rules []*SGRule, isIngress bool, connectivityMap map[*ipblock.IPBlock]*ConnectivityResult) {
+	Locals := []*ipblock.IPBlock{}
+	for i := range rules {
+		if rules[i].local != nil {
+			Locals = append(Locals, rules[i].local)
+		}
+	}
+	disjointLocals := ipblock.DisjointIPBlocks(Locals, []*ipblock.IPBlock{ipblock.GetCidrAll()})
+	for i := range disjointLocals {
+		connectivityMap[disjointLocals[i]] = AnalyzeSGRules(rules, isIngress)
+	}
```

Comment:
it seems that this will assign the same result to each disjoint local entry in the map, since each time ` AnalyzeSGRules` is called with the same set of rules, instead of considering only the relevant rules for this local entry